pkg/csi/service/osutils/linux_os_utils.go

```diff
@@ -832,6 +849,138 @@ func (osUtils *OsUtils) GetDevMounts(ctx context.Context,
 	return devMnts, nil
 }
 
+// getMountInfo scans through /proc/self/mountinfo and
+// returns complete information for each entry in the file.
+func getMountInfo(ctx context.Context) ([]MountInfoDetail, error) {
+	infos := make([]MountInfoDetail, 0)
+	content, err := readProcsFile(procMountsPath)
+	if err != nil {
+		return infos, err
+	}
+	buffer := bytes.NewBuffer(content)
+	infos, err = readProcMountsFrom(ctx, buffer, true)
+	return infos, err
+}
+
+// readProcMountsFrom reads through the given file
+// and construct MountInfo struct objects.
+func readProcMountsFrom(ctx context.Context,
```

Comment:
Can we see if we can utilize existing lib from `k8s.io/mount-utils` ?
The goal is to remove local functions for mount-related logic and rely entirely on `k8s.io/mount-utils`